components/org.wso2.carbon.identity.api.server.organization.management/org.wso2.carbon.identity.api.server.organization.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/management/v1/service/OrganizationManagementService.java

```diff
@@ -915,6 +943,61 @@ private OrganizationMetadata getOrganizationMetadataResponse(Organization organi
         return organizationMetadata;
     }
 
+    private MetaAttributesResponse getMetaAttributesResponse(Integer limit, String after, String before, String filter,
+                                                             List<String> metaAttributes, boolean recursive)
+            throws OrganizationManagementServerException {
+
+        MetaAttributesResponse metaAttributesResponse = new MetaAttributesResponse();
+
+        if (limit != 0 && CollectionUtils.isNotEmpty(metaAttributes)) {
+            boolean hasMoreItems = metaAttributes.size() > limit;
+            boolean needsReverse = StringUtils.isNotBlank(before);
+            boolean isFirstPage = (StringUtils.isBlank(before) && StringUtils.isBlank(after)) ||
+                    (StringUtils.isNotBlank(before) && !hasMoreItems);
+            boolean isLastPage = !hasMoreItems && (StringUtils.isNotBlank(after) || StringUtils.isBlank(before));
+
+            String url = "?limit=" + limit + "&recursive=" + recursive;
+            if (StringUtils.isNotBlank(filter)) {
+                try {
+                    url += "&filter=" + URLEncoder.encode(filter, StandardCharsets.UTF_8.name());
+                } catch (UnsupportedEncodingException e) {
+                    LOG.error("Server encountered an error while building pagination URL for the response.", e);
+                    Error error = OrganizationManagementEndpointUtil.getError(
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getCode(),
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getMessage(),
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getDescription());
+                    throw new OrganizationManagementEndpointException(Response.Status.INTERNAL_SERVER_ERROR, error);
```

Comment:
Can't we throw this as a OrganizationManagementServerException and then this will be handled in https://github.com/wso2/identity-api-server/pull/628/files#diff-09c18939b768ab874a9ba6be6e8d751bafa0398cae65158fe4d5ab8774a60560R569.

Reply:
This is now updated in all places.